cli/interaction.ss

```diff
@@ -257,31 +262,72 @@
              (option 'timeout-in-blocks "-T" "--timeout-in-blocks" default: #f
                      help: "number of blocks after which to time out")
              (option 'handshake "-H" "--handshake" default: #f
-                     help: "command to use to transfer handshakes")]
+                     help: "command to use to transfer handshakes")
+             ;; TODO: Abstract into Enum - See gerbil-poo
+             ;; enum off-chain-channel = 'stdio | 'libp2p
+             (option 'off-chain-channel-selection "-C" "--off-chain-channel" default: 'stdio
+                     help: "command to specify off-chain-channel")
+             (option 'host-address "-O" "--host-address" default: #f
+                     help: "host-address (only required if using libp2p as off-chain-channel)")
+             ;; FIXME: This should be stored and extracted from `dest-address`.
+             ;; It is supplied here as a temporary workaround.
+             (option 'dest-address "-D" "--dest-address" default: #f
+                     help: "dest-address (only required if using libp2p as off-chain-channel)")
+             (flag 'wait-for-agreement "-W" "--wait-for-agreement"
+                   help: "wait for agreement via off-chain-channel")]
             [(lambda (opt) (hash-remove! opt 'test))]
             [options/glow-path options/contacts
              options/evm-network options/database options/test options/backtrace]))
+
+  ;; NOTE: This also populates the runtime nickname-address mapping table.
+  (def contacts (load-contacts contacts-file))
+  ;; TODO: Validate whether you possess the necessary keys to assume this identity.
+  ;; TODO: Remove request for identity at stdio site.
+  ;; 1. Update integration tests to answer prompts earlier,
+  ;; 2. before agreements are generated / consumed (start-interaction/generate-agreement / start-interaction/with-agreement).
+  ;; Once 1, 2 are done, we can remove the conditional for this.
+  (def my-nickname
+    (match off-chain-channel-selection
+      ("libp2p" (or identity (prompt-identity)))
+      (else identity)))
+
   (def options
        (hash
          (glow-app glow-app)
-         (identity identity)
+         (identity my-nickname)
          (params (string->json-object (or params "{}")))
          (participants (string->json-participant-map (or participants "{}")))
          (assets (string->json-asset-map (or assets "{}")))
          (max-initial-block max-initial-block)
          (timeout-in-blocks timeout-in-blocks)
          (role role)))
+
+  ;; TODO: abstract into Poo object, especially if we have more local-runtime-options
```

Comment:
Leaving this out first, since we still need still store `dest-address` into `contacts`, so the `channel-options` schema will change.